core/src/main/java/com/datastrato/gravitino/storage/kv/TransactionalKvBackendImpl.java

```diff
@@ -124,8 +124,8 @@ public void rollback() throws IOException {
 
   @Override
   public void closeTransaction() {
-    putPairs.get().clear();
-    originalKeys.get().clear();
+    putPairs.remove();
```

Comment:
I am not sure why were not removing it before? I think @yuqi1129 can have some inputs here.

Reply:
Indeed, removing the object is safer than just clearing it.